### DISCUSSION TITLE
Modal Analysis

### DISCUSSION THREAD

**Discussion by Qaslu** [petrobras/ross#937] [Q&A]
Hi, 
I just started diving into the field of rotordynamics and using ross, so my apologies if my question is somewhat basic. 
When using the function run_modal, I fail to grasp the influence of the required input speed. In most examples I could find and in the ross documentation it is usually set to zero. It also seems, except for very high values, to have little influence on the calculated damped and undamped natural frequencies. 
When plotting the mode shape using either plot_mode_2d or plot_mode_3d, the frequency/speed is again required in the form of the n'th input mode. 
I guess my questions are: what exactly is the input speed for the run_modal function, and how is it selected? 
Thanks a lot for your help!

**Comment by raphaeltimbo** [comment]
Hi @Qaslu !
The speed argument in the `.run_modal()` method is used to calculate the bearings/seals stiffness and damping coefficients. These coefficients, for real machines, are normally speed dependent, so the speed will affect the calculation of the rotor natural frequencies. 

A nice way to visualize this is the Campbell Diagram. We have an example in the tutorial with a real machine [here](https://ross.readthedocs.io/en/stable/tutorials/tutorial_part_2.html#plotting-results-a-id-section1-3-2-a).

The speed will also have an impact in the gyroscopic effect. A rotor with an overhung disk is greatly affected by this as shown in [this example] (https://ross.readthedocs.io/en/stable/discussions/example_8.html), where you can see the modes splitting into forward and backward as we increase the speed.

**Reply by FadliM21** [reply]
Hi @raphaeltimbo ,
I couldn't open the shared link above, may I have access to the link regarding the example with a real machine files?

Thank you.
Regards,
Fadli

**Reply by ViniciusTxc3** [reply]
@FadliM21 Here are the links to access:

Campbell diagram: [In 1.3.2](https://ross.readthedocs.io/en/latest/user_guide/tutorial_part_2.html)
Gyroscopic effect: [Example 8](https://ross.readthedocs.io/en/latest/user_guide/example_8.html)

**Reply by FadliM21** [reply]
Thanks @ViniciusTxc3 , 

May I ask another problem? 
I've tried to simulate with real case our machine here, but I still have different result with the Vendor's calculation (they used Madyn 2000), 
For the 1st critical speed, the shape mode is similar although still has difference in value. But, for the 2nd critical speed, I couldn't see it in the Ross Campbell diagram while the vendor's result has the 2nd critical speed on 2x speed mode.
Could you please help me to solve this case?

Thank you in advance. I really appriciate your kindly response.

Regards,
Fadli

**Reply by ViniciusTxc3** [reply]
Hi @FadliM21 !
I can't see your problem, could you provide the code you are running so I can try to identify the problem?

**Reply by FadliM21** [reply]
[Model_Rotor_234.zip](https://github.com/petrobras/ross/files/15502442/Model_Rotor_234.zip)

Hi @ViniciusTxc3 ,
Thanks for your response, Please find attached for the code that I've tried and some of Vendor's reference.

Thank you very much for your kindly helps.
regards,
Fadli

**Reply by ViniciusTxc3** [reply]
@FadliM21 this topic will be addressed in the #1041, facilitating future discussions.

**Reply by FadliM21** [reply]
Hi @ViniciusTxc3 ,
in the Campbell diagram, Can I change the X Axis (Rotor Speed) in RPM while the Y axis (Frequency) still keep in Hz?

Thanks
Regards,
Fadli

**Reply by ViniciusTxc3** [reply]
@FadliM21 Currently, it is only possible to have the same unit for the X and Y axis, using the parameter `frequency_units`. As the example:

```pyhton
import ross as rs
import numpy as np
Q_ = rs.Q_
rotor = rs.rotor_example()
speed = np.linspace(0, 400, 101)
camp = rotor.run_campbell(speed)
fig = camp.plot(
             harmonics=[1, 2],
             damping_parameter="damping_ratio",
             frequency_range=Q_((2000, 10000), "RPM"),
             damping_range=(-0.1, 100),
             frequency_units="RPM",
        )
```

I'll create an issue to evaluate the possibility of implementing this change.
